### ISSUE TITLE
Reassigning "export" global breaks CJS imports

### ISSUE THREAD

**Issue by nabetti1720** [awslabs/llrt#642]
It appears that LLRT is unable to return information to the caller if the module loaded by require has the following implementation.　This implementation can be seen in the browser-oriented implementation of the Node.js package called `debug`.

```javascript
// require10.js
exports = module.exports;

exports.str = "str";
exports.cat = function cat() {
  return exports.str;
}

exports.array = [];
exports.length = function length() {
  return exports.array.length;
}
```

```javascript
// index10.js
const a = require("./require10.js")

console.log(a.cat());
console.log(a.length());
```

```
% llrt index10.js
undefined
TypeError: cannot read property 'length' of undefined
  at length (/Users/shinya/Workspaces/llrt-test/./require10.js:10:24)
      at <anonymous> (index10.js:4:15)
      at <anonymous> (eval_script:1:5)

% bun index10.js 
str
0
```

**Comment by richarddavison** [comment]
> It appears that LLRT is unable to return information to the caller if the module loaded by require has the following implementation.　This implementation can be seen in the browser-oriented implementation of the Node.js package called `debug`.
> 
> ```js
> // require10.js
> exports = module.exports;
> 
> exports.str = "str";
> exports.cat = function cat() {
>   return exports.str;
> }
> 
> exports.array = [];
> exports.length = function length() {
>   return exports.array.length;
> }
> ```
> 
>  ```js
> // index10.js
> const a = require("./require10.js")
> 
> console.log(a.cat());
> console.log(a.length());
> ```
> 
>  ```
> % llrt index10.js
> undefined
> TypeError: cannot read property 'length' of undefined
>   at length (/Users/shinya/Workspaces/llrt-test/./require10.js:10:24)
>       at <anonymous> (index10.js:4:15)
>       at <anonymous> (eval_script:1:5)
> 
> % bun index10.js 
> str
> 0
> ```

This is a tricky one. With the example above we’re redefining the global variable exports with our own one. The global variable exports in LLRT stores exports via an accessor. This is then used to return the exported props. Will take a look.

**Comment by nabetti1720** [comment]
Thank you for addressing this issue. I would add that `exports = module.exports;` does not seem to be related to this issue. It works as expected for Node.js and Bun with or without this code.

```javascript
// require10.cjs
//exports = module.exports;

exports.str = "str";
exports.cat = function cat() {
  return exports.str;
}

exports.array = [];
exports.length = function length() {
  return exports.array.length;
}
```

```
// index10.cjs
const a = require("./require10.cjs")

console.log(a.cat());
console.log(a.length());
```

```
% llrt index10.cjs 
undefined
TypeError: cannot read property 'length' of undefined
  at length (/Users/shinya/Workspaces/llrt-test/./require10.cjs:10:24)
      at <anonymous> (index10.cjs:4:15)
      at <anonymous> (eval_script:1:5)

% node index10.cjs 
str
0

% bun index10.cjs
str
0
```

**Comment by richarddavison** [comment]
Sorry for not being clear. That's probably the root cause of this bug in LLRT, it works as expected in bun and deno :)

**Comment by nabetti1720** [comment]
Closed as fixed in #651.
